pkg/api/write.go

```diff
@@ -45,23 +48,10 @@ func Write(writer pgmodel.DBInserter, elector *util.Elector, metrics *Metrics) h
 
 		metrics.LeaderGauge.Set(1)
 
-		compressed, err := ioutil.ReadAll(r.Body)
-		if err != nil {
-			log.Error("msg", "Read error", "err", err.Error())
-			http.Error(w, err.Error(), http.StatusInternalServerError)
-			return
-		}
-
-		reqBuf, err := snappy.Decode(nil, compressed)
+		req, err, logMsg := loadWriteRequest(r)
+		defer r.Body.Close()
```

Comment:
defers usually go after the err check. Is this a special case?

Reply:
Actually, this is my bad, we don't need to close it for requests. Will remove.